src/useScript/useScript.ts

```diff
@@ -1,67 +1,115 @@
+        setStatus(cachedScriptStatus)
         return
       }
 
       // Fetch existing script element by src
       // It may have been added by another instance of this hook
-      let script: ScriptElt = document.querySelector(`script[src="${src}"]`)
+      const script = getScriptNode(src)
+      let scriptNode = script.node
 
-      if (!script) {
-        // Create script
-        script = document.createElement('script')
-        script.src = src
-        script.async = true
-        script.setAttribute('data-status', 'loading')
-        // Add script to document body
-        document.body.appendChild(script)
+      if (!scriptNode) {
+        // Create script element and add it to document body
+        scriptNode = document.createElement('script')
+        scriptNode.src = src
+        scriptNode.async = true
+        scriptNode.setAttribute('data-status', 'loading')
+        document.body.appendChild(scriptNode)
 
         // Store status in attribute on script
         // This can be read by other instances of this hook
         const setAttributeFromEvent = (event: Event) => {
-          script?.setAttribute(
-            'data-status',
-            event.type === 'load' ? 'ready' : 'error',
-          )
+          const scriptStatus: UseScriptStatus =
+            event.type === 'load' ? 'ok' : 'error'
+
+          scriptNode?.setAttribute('data-status', scriptStatus)
         }
 
-        script.addEventListener('load', setAttributeFromEvent)
-        script.addEventListener('error', setAttributeFromEvent)
+        scriptNode.addEventListener('load', setAttributeFromEvent)
+        scriptNode.addEventListener('error', setAttributeFromEvent)
       } else {
         // Grab existing script status from attribute and set to state.
-        setStatus(script.getAttribute('data-status') as Status)
+        setStatus(script.status ?? cachedScriptStatuses[src] ?? 'loading')
       }
 
       // Script event handler to update status in state
       // Note: Even if the script already exists we still need to add
       // event handlers to update the state for *this* hook instance.
       const setStateFromEvent = (event: Event) => {
-        setStatus(event.type === 'load' ? 'ready' : 'error')
+        const newStatus = event.type === 'load' ? 'ok' : 'error'
+        setStatus(newStatus)
+        cachedScriptStatuses[src] = newStatus
       }
 
       // Add event listeners
-      script.addEventListener('load', setStateFromEvent)
-      script.addEventListener('error', setStateFromEvent)
+      scriptNode.addEventListener('load', setStateFromEvent)
+      scriptNode.addEventListener('error', setStateFromEvent)
 
       // Remove event listeners on cleanup
       return () => {
-        if (script) {
-          script.removeEventListener('load', setStateFromEvent)
-          script.removeEventListener('error', setStateFromEvent)
+        if (scriptNode && removeOnUnmount) {
```

Comment:
My original intent was to have `removeOnUnmount` actually remove the `scriptNode` from the DOM. I can't think of a use case where we wouldn't want to `removeEventListener`.

https://github.com/juliencrn/usehooks-ts/pull/223/files#diff-6ba736bc4c679006c0e6aac2fc18386cb2b6e5a371ff12712ed03fc03c7176f1R68-R70

Reply:
Indeed, my bad. Good catch! Thanks

Reply:
Yeah! No problem at all.